app/requests/remote/user.ts

```diff
@@ -105,27 +103,39 @@ export const login = async ({ldapOnly = false, loginId, mfaToken, password}: Log
 };
 
 export const loadMe = async ({deviceToken, user}: LoadMeArgs) => {
-    let currentUser: RawUser = user;
+    let currentUser = user ?? undefined;
 
     const {activeServerDatabase, error} = await getActiveServerDatabase();
     if (!activeServerDatabase) {
-        return {error};
+        return {
+            error,
+            currentUser: undefined,
+        };
     }
 
     try {
         if (deviceToken) {
             await Client4.attachDevice(deviceToken);
         }
 
-        if (!user) {
+        if (!currentUser) {
             currentUser = ((await Client4.getMe()) as unknown) as RawUser;
         }
     } catch (e) {
         await forceLogoutIfNecessary(e);
-        return {error: e};
+        return {
+            error: e,
+            currentUser: undefined,
+        };
     }
 
     try {
+        if (!currentUser) {
```

Comment:
why it wouldn't be defined here? or is this just TS being safe? if it is TS, then you don't need to return in the above catch.

You can leave this as is as I'm going to refactor this code with the entry point logic

Reply:
With the SSO login flow, we call this `loadMe` method without a user profile.  At this point, we check to see if the `currentUser` is indeed `undefined` before fetching its detail from the server via the `Client.getMe()` call.